laevatein/src/main/java/com/laevatein/internal/utils/MediaStoreUtils.java

```diff
@@ -0,0 +1,343 @@
+import java.util.ArrayList;
+import java.util.Date;
+
+/**
+ * Compatibility class for the issue of various implementations of camera feature.
+ * <p>
+ * This class is based on this source code.
+ * https://github.com/mixi-inc/Android-Device-Compatibility/blob/master/AndroidDeviceCompatibility/src/main/java/jp/mixi/compatibility/android/provider/MediaStoreCompat.java
+ * <p>
+ * License
+ * see: https://github.com/mixi-inc/Android-Device-Compatibility#license
+ */
+public class MediaStoreUtils {
+    public static final String TAG = MediaStoreUtils.class.getSimpleName();
+    private static final String MEDIA_FILE_NAME_FORMAT = "yyyyMMdd_HHmmss";
+    private static final String MEDIA_FILE_EXTENSION = ".jpg";
+    private static final String MEDIA_FILE_PREFIX = "IMG_";
+    private final String MEDIA_FILE_DIRECTORY;
+    private Context mContext;
+    private ContentObserver mObserver;
+    private ArrayList<PhotoContent> mRecentlyUpdatedPhotos;
+
+    /**
+     * Prepares to deal with various implementations of camera feature and {@link MediaStore}.
+     * You should call {@link MediaStoreUtils#destroy()} on destroying context.
+     *
+     * @param context a context
+     * @param handler a handler
+     */
+    public MediaStoreUtils(Context context, Handler handler) {
+        mContext = context;
+        MEDIA_FILE_DIRECTORY = context.getPackageName();
+        mObserver = new ContentObserver(handler) {
+            @Override
+            public void onChange(boolean selfChange) {
+                super.onChange(selfChange);
+                updateLatestPhotos();
+            }
+        };
+        mContext.getContentResolver().registerContentObserver(
+                MediaStore.Images.Media.EXTERNAL_CONTENT_URI, true, mObserver);
+    }
+
+    /**
+     * Checks whether the device has a camera feature or not.
+     *
+     * @param context a context to check for camera feature.
+     * @return true if the device has a camera feature. false otherwise.
+     */
+    public static final boolean hasCameraFeature(Context context) {
+        PackageManager pm = context.getApplicationContext().getPackageManager();
+        return pm.hasSystemFeature(PackageManager.FEATURE_CAMERA);
+    }
+
+    /**
+     * Invokes a camera capture activity.
+     *
+     * @param activity    the caller of the camera capture activity.
+     * @param requestCode activity result handling id.
+     * @return a file name to be saved as.
+     */
+    public String invokeCameraCapture(Activity activity, int requestCode) {
+        if (!hasCameraFeature(mContext)) return null;
+
+        File toSave = getOutputFile();
+        if (toSave == null) return null;
+
+        Intent intent = new Intent(MediaStore.ACTION_IMAGE_CAPTURE);
+        intent.addCategory(Intent.CATEGORY_DEFAULT);
+        intent.putExtra(MediaStore.EXTRA_OUTPUT, FileProvider.getUriForFile(activity, activity.getString(R.string.l_file_provider_authorities), toSave));
```

Comment:
This is different from MediaStoreCompat.
see: https://github.com/mixi-inc/Android-Device-Compatibility/blob/master/AndroidDeviceCompatibility/src/main/java/jp/mixi/compatibility/android/provider/MediaStoreCompat.java#L108